modus/agents.mdx

```diff
@@ -0,0 +1,755 @@
+---
+title: "What is an Agent?"
+description: "Learn about stateful agents in Modus"
+"og:title": "What is an Agent? - Modus"
+---
+
+## Agents in Modus
+
+Agents in Modus are elite operatives that maintain persistent memory across
+missions. Unlike stateless functions that lose everything when operations end,
+agents remember every detail, survive system failures, and never lose their
+operational context.
+
+## Key characteristics
+
+- **Stateful**: Maintains memory and context across interactions
+- **Persistent**: Automatically saves and restores state
+- **Resilient**: Graceful recovery from failures
+- **Autonomous**: Can operate independently over extended periods
+- **Actor-based**: Each agent instance runs in isolation
+- **Event-driven**: Streams real-time updates and operational intelligence
+
+## When to deploy agents
+
+Agents are perfect for:
+
+- **Multi-turn workflows** spanning multiple interactions
+- **Long-running processes** that maintain context over time
+- **Stateful operations** that need to remember previous actions
+- **Complex coordination** between different system components
+- **Persistent monitoring** that tracks changes over time
+- **Real-time operations** requiring live status updates and event streaming
+
+## Agent structure
+
+Every agent starts with the essential operational framework:
+
+```go
+package main
+
+import (
+    "fmt"
+    "strings"
+    "time"
+    "github.com/hypermodeinc/modus/sdk/go/pkg/agents"
+    "github.com/hypermodeinc/modus/sdk/go/pkg/models"
+    "github.com/hypermodeinc/modus/sdk/go/pkg/models/openai"
+)
+
+type IntelligenceAgent struct {
+    agents.AgentBase
+
```

Comment:
add a comment here indicating that the rest of the fields make up the agent's state and can be customized per agent

Reply:
* https://github.com/hypermodeinc/docs/pull/144/commits/5eb42efb64c39379ab63b0335f21417225c1f115